README.md

```diff
@@ -1,2 +1,105 @@
-# inverted-index
-Javascript Checkpoint 1: Develop an inverted index object that takes a JSON array of text objects and creates an index from the array. The index allows a user to search for text blocks in the array that contain a specified collection of words.
+# Andela Javascript Checkpoint 1: inverted-index
+## Description
+Develop an [inverted index] (https://www.elastic.co/blog/found-elasticsearch-from-the-bottom-up) object that takes a JSON array of text objects and creates an index from the array. The index allows a user to search for text blocks in the array that contain a specified collection of words.
+
+## Methodology
+### createIndex(filePath)
+This Method accepts a filePath as its argument and creates an inverted-index from the input JSON file. Each word in the inverted-index is matched to its source document and its individual position in its source object.
+
+### getIndex()
+This Method returns the result of the createIndex method, an object that has the format:
+```javaScript
+{ word : [[documentID, wordPositionID]],
+  word2 : [[documentID, wordPositionID],[documentId, wordPositionID]]}
+ ```
+`documentID` refers to the numerical identity of the document in which the word is found.
+`wordPositionID` refers to the poistion of the word in the document it is found.
+
+### searchIndex(term)
+This Method takes a `term` as an argument and filters the index for the term, returning the `documentID` and `wordPositionID` for the term:
+```javaScript
+[documentId, wordPositionID]
+```
+
+### getFrequency(term)
+This method takes a `term` as an argument and returns a value repersenting how many times the 'term' appears in the input file.
+
+## Functionality
+#### Create an instance of the Index Class and call the createIndex and getIndex methods, for example:
+
+```javaScript
+var _index = new Index();
+_index.createIndex('/jasmine/books.json').then(function() {
+  console.log(_index.getIndex());
+
+  // Search for these 'terms'
+  console.log(_index.searchIndex('alice'));
+  console.log(_index.searchIndex('and'));
+  console.log(_index.searchIndex('rudyard'));
+  console.log(_index.searchIndex('astronomy'));
+  console.log(_index.searchIndex('wonderland'));
+  
+  // Will it throw an invalid type error for non-string type entries?
+  console.log(_index.searchIndex(090));`
+  console.log(_index.searchIndex(true));
+});
+```
+
+**For the books.JSON test objects first document, alice, the result would be:**
+
+```javaScript
+   { title : [0 , 0],
+	alice : [[0 , 1],[0 , 5]],
```

Comment:
**Code Quality - Nitpick**

It is difficult to understand the structure of your index given the other parts you have added for word position. An array isn't the best option to show this info especially if a document is nested.